torchtitan/components/dataloader.py

```diff
@@ -41,6 +41,7 @@ def __iter__(self):
         ...
 
 
+# pyrefly: ignore [inconsistent-inheritance]
```

Comment:
TODO for me: this seems to be an inconsistency issue between checkpoint Stateful and dataloader StatefulDataLoader, which we may not be able to change.